protocol/dubbo/client.go

```diff
@@ -267,11 +268,25 @@ func (c *Client) call(ct CallType, request *Request, response *Response, callbac
 		return errSessionNotExist
 	}
 	defer func() {
+		failNumber := 0
 		if err == nil {
-			c.pool.put(conn)
-			return
+			for {
+				ok := atomic.CompareAndSwapUint32(&c.pool.pushing, 0, 1)
+				if ok {
+					c.pool.poolQueue.pushHead(conn)
+					c.pool.pushing = 0
```

Comment:
Need we use `atomic` here?

Reply:
yes, pushHead adds val at the head of the queue. It must only be called by a single producer.